src/app/work-item/work-item-list/work-item-list.component.ts

```diff
@@ -248,4 +248,45 @@ export class WorkItemListComponent implements OnInit, AfterViewInit {
     this.showTypesOptions = false;
     this.router.navigate(['/work-item-list/detail/new?' + type]);
   }
-}
\ No newline at end of file
+
+  onDragStart() {
+    console.log('on drag start');
+  }
+
+  //Event listener for WI drop.
+  onDragEnd(workItemId: number) {
+    let droppedElement = document.getElementById('work-item-'+workItemId);
+    let previousElement = droppedElement.previousElementSibling;
+    let nextElement = droppedElement.nextElementSibling;
+    let prevItemId: string;
+    let nextItemId: string;
+
+    if(previousElement) {
+      prevItemId = previousElement.getAttribute('data-id');
+    }
+    else {
+      prevItemId = '';
+    }
+    if(nextElement) {
+      nextItemId = nextElement.getAttribute('data-id');
+    }
+    else {
+      nextItemId = '';
+    }
+
+    this.workItemService
```

Comment:
There should not be any calculation in this portion. It should happen inside the service only. In case @vikram-raj the re ordering on the big list of work item is not needed because his library is already taking a reference of the source array and re ordering that. Only buildWorkItemIdIndexMap should do. In case of move top or move bottom the array needs to be rearranged as well. @nimishamukherjee let's talk about that.